stan/math/rev/core/chainablestack.hpp

```diff
@@ -11,6 +11,18 @@ class chainable_alloc;
 
 typedef AutodiffStackSingleton<vari, chainable_alloc> ChainableStack;
 
+/**
+ * Instantiates an instance of the ChainableStack if not already
+ * initialized. This function must be called before any autodiff
+ * variable get's instantiated within any thread which performs
+ * reverse mode autodiff operations.
+ */
+static inline ChainableStack::AutodiffStackStorage* init() {
```

Comment:
Should we give this a slightly longer name indicating when you need to call it? Maybe something like `init_thread_ad()`?

Reply:
Hmm... my thoughts:

- It shoould not have "thread" in it's name. After all you must call this in a non-threaded application as well (imagine someone does not rely on our makefiles then the call it `init` is not automatically included in the code).
- We never refer to "ad" in our function names at all. So this is not a convention we use.

If you want it more verbose, then how about `init_stack()` or `init_tape()`? I prefer stack as this is the naming used in stan-math.